frontend/src/pages/application/Settings.vue

```diff
@@ -71,26 +72,51 @@ export default {
     emits: ['application-delete', 'application-updated'],
     data () {
         return {
-            sideNavigation: [{ name: 'General', path: './settings' }],
+            sideNavigation: [{
+                name: 'General',
+                path: './settings'
+            }],
             input: {
                 projectName: this.application.name,
-                projectId: this.application.id
+                projectId: this.application.id,
+                application: this.application
+            },
+            editing: false,
+            options: {
+                instances: []
             },
-            editing: false
+            loading: {
+                instances: false
+            }
         }
     },
     computed: {
         formValid () {
             return this.input.projectName
+        },
+        getDeleteApplicationText () {
+            if (this.options.instances.length === 0) {
+                return 'Once deleted, your application  permanently deleted. This cannot be undone.'
+            } else {
+                return 'Once you delete all your instances of this application, you can delete this application.'
+            }
         }
     },
-    watch: {
-        project () {
-            this.input.projectName = this.application.name
-            this.input.projectId = this.application.id
-        }
+    mounted () {
+        this.loadInstances(this.application.id)
     },
     methods: {
+        loadInstances (applicationId) {
+            this.loading.instances = true
+            ApplicationAPI.getApplicationInstances(applicationId)
+                .then((instances) => {
+                    this.options.instances = instances || []
```

Comment:
Given that all we care about here is the number of instances, it's worth just doing `this.options.instances = instances?.length || 0` to save having to store the whole `instances` array in memory.

Then, on `line 36` you wouldn't need the `.length`.

Reply:
Thank you for the suggestion! That's a great optimization idea. Since we only care about the number of instances, storing the length of the instances array instead of the whole array itself would save memory. I will make the changes to implement this optimization